### PR TITLE
Add okd-4.1.0-rc.0 cluster provider

### PR DESCRIPTION
**Pull request by cynepco3hahue**
**What this PR does / why we need it**:
Provides the possibility to run and test KubeVirt on top of 4.1.0-rc.0 OKD cluster.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add okd-4.1.0-rc.0 cluster provider
```

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
grrr I hate dynamic linking
```
+ oc adm taint nodes test-1-w9d92-master-0 node-role.kubernetes.io/master-
node/test-1-w9d92-master-0 untainted
cluster/okd-4.0.9/.kubectl: error while loading shared libraries: libcrypto.so.10: cannot open shared object file: No such file or directory
Makefile:104: recipe for target 'cluster-up' failed
make: *** [cluster-up] Error 127
```
@rmohr I am thinking to create something like `gocli kubectl` command, that will run all `kubectl` commands inside of  cluster container, maybe do you have some better idea how to solve this dynamic dependency on the `libcrypto.so.10`

File: cluster/ephemeral-provider-common.sh [hunk 1]
```diff
@@ -46,19 +46,6 @@ function build() {
     ${KUBEVIRT_PATH}hack/dockerized "DOCKER_PREFIX=${DOCKER_PREFIX} DOCKER_TAG=${DOCKER_TAG} KUBEVIRT_PROVIDER=${KUBEVIRT_PROVIDER} IMAGE_PULL_POLICY=${IMAGE_PULL_POLICY} VERBOSITY=${VERBOSITY} ./hack/build-manifests.sh"
     ${KUBEVIRT_PATH}hack/dockerized "DOCKER_PREFIX=${DOCKER_PREFIX} DOCKER_TAG=${DOCKER_TAG} KUBEVIRT_PROVIDER=${KUBEVIRT_PROVIDER} ./hack/bazel-build.sh"
     ${KUBEVIRT_PATH}hack/dockerized "DOCKER_PREFIX=${DOCKER_PREFIX} DOCKER_TAG=${DOCKER_TAG} DOCKER_TAG_ALT=${DOCKER_TAG_ALT} KUBEVIRT_PROVIDER=${KUBEVIRT_PROVIDER} ./hack/bazel-push-images.sh"
-
-    # Make sure that all nodes use the newest images
```
Comment (davidvossel) [inline cluster/ephemeral-provider-common.sh]: i thought we still need this?

Comment (cynepco3hahue) [reply cluster/ephemeral-provider-common.sh]: To be honest, I do not sure, I remember we had some issues with it, but if we had `pullPolicy: Always` for the dev environment, it should get new images on each `make cluster-sync`, also now with podman and cri-o you cannot be sure that the node will have docker at all.
@rmohr Do you remember why do we have it?

Comment (davidvossel) [reply cluster/ephemeral-provider-common.sh]: I believe it was so we'd have the images pre-cached on the nodes.  If CI passes with this removed, then i doubt it's necessary anymore.

Comment (cynepco3hahue) [reply cluster/ephemeral-provider-common.sh]: I can see jobs run without any troubles.

**Comment by cynepco3hahue** [comment]
@davidvossel @slintes Provider is ready for use, but it will need much more investigation why we have a lot of tests that fail on it :worried: 

That the list of failures that I had locally:
```
Summarizing 31 Failures:

[Fail] Storage Starting a VirtualMachineInstance With a HostDisk defined With 'DiskExistsOrCreate' type [It] Should create a disk image and start 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Storage Starting a VirtualMachineInstance With a HostDisk defined With 'DiskExists' type [BeforeEach] Should use existing disk image and start 
/root/go/src/kubevirt.io/kubevirt/tests/storage_test.go:361

[Fail] Storage Starting a VirtualMachineInstance With smaller than requested PVCs [It] Should not initialize an empty PVC with a disk.img when disk is too sm
all even with toleration 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Storage Starting a VirtualMachineInstance With smaller than requested PVCs [It] Should initialize an empty PVC with a disk.img when disk is too small 
but within toleration 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Genie VirtualMachineInstance with cni flannel and ptp plugin interface [It] should create a virtual machine with one interface 
/root/go/src/kubevirt.io/kubevirt/tests/vmi_genie_test.go:83

[Fail] Genie VirtualMachineInstance with cni flannel and ptp plugin interface [It] should create a virtual machine with two interfaces 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:319

[Fail] Migrations Starting a VirtualMachineInstance  with a Cirros disk [It] should be successfully migrated multiple times with cloud-init disk 
/root/go/src/kubevirt.io/kubevirt/tests/migration_test.go:177

[Fail] Migrations Starting a VirtualMachineInstance  with a shared ISCSI Filesystem PVC [It] should migrate a vmi with UNSAFE_MIGRATION flag set 
/root/go/src/kubevirt.io/kubevirt/tests/migration_test.go:352

[Fail] Migrations Starting a VirtualMachineInstance  with an Alpine DataVolume [It] should reject a migration of a vmi with a non-shared data volume 
/root/go/src/kubevirt.io/kubevirt/tests/migration_test.go:398

[Fail] Migrations Starting a VirtualMachineInstance  migration security [It] should secure migrations with TLS 
/root/go/src/kubevirt.io/kubevirt/tests/migration_test.go:655

[Fail] Migrations with sata disks [It] VM with containerDisk + CloudInit + ServiceAccount + ConfigMap + Secret 
/root/go/src/kubevirt.io/kubevirt/tests/migration_test.go:177

[Fail] Migrations with a live-migrate eviction strategy set with a VMI running with an eviction strategy set [It] should block the eviction api while a slow migration is in progress 
/root/go/src/kubevirt.io/kubevirt/tests/migration_test.go:1011

[Fail] [rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:component]Expose Expose service on a VMI replica set [BeforeEach] Expose ClusterIP service [test_id:1537]Should create a ClusterIP service on VMRS and connect to it
/root/go/src/kubevirt.io/kubevirt/tests/expose_test.go:50

[Fail] Multus [rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance using different types of interfaces. VirtualMachineInstance with cni ptp plugin interface [It] [test_id:1752]should create a virtual machine with one interface with network definition from different namespace 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Multus [rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance using different types of interfaces. VirtualMachineInstance with ovs-cni plugin interface [It] [test_id:1577]should create two virtual machines with one interface 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Multus [rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance using different types of interfaces. VirtualMachineInstance with ovs-cni plugin interface [It] [test_id:1578]should create two virtual machines with two interfaces 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Multus [rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance using different types of interfaces. VirtualMachineInstance with ovs-cni plugin interface and custom MAC address. [It] [test_id:676]should configure valid custom MAC address on ovs-cni interface. 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Multus [rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance using different types of interfaces. Single VirtualMachineInstance with ovs-cni plugin interface [It] [test_id:1756]should report all interfaces in Status 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Multus [rfe_id:1758][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance definition with quemu guest agent [It] [test_id:1757] should report guest interfaces in VMI status 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Configurations [rfe_id:897][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance with CPU pinning with cpu pinning enabled [It] [test_id:1684]should set the cpumanager label to false when it's not running 
/root/go/src/kubevirt.io/kubevirt/tests/vmi_configuration_test.go:1349

[Fail] Configurations [rfe_id:897][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance with CPU pinning with cpu pinning enabled [It] [test_id:1685]non master node should have a cpumanager label 
/root/go/src/kubevirt.io/kubevirt/tests/vmi_configuration_test.go:1372

[Fail] Configurations [rfe_id:897][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance with CPU pinning with cpu pinning enabled [It] [test_id:1686]should be scheduled on a node with running cpu manager 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Configurations [rfe_id:897][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance with CPU pinning with cpu pinning enabled [It] [test_id:1687]should configure correct number of vcpus with requests.cpus 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] Configurations [rfe_id:897][crit:medium][vendor:cnv-qe@redhat.com][level:component]VirtualMachineInstance with CPU pinning with cpu pinning enabled [It] [test_id:1691]should start a vm with no cpu pinning after a vm with cpu pinning on same node 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:344

[Fail] ImageUpload Upload an image and start a VMI [It] Should succeed 
/root/go/src/kubevirt.io/kubevirt/tests/imageupload_test.go:67

[Fail] [rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:component]Networking Multiple virtual machines connectivity should be able to reach [It] [test_id:1541]the Inbound VirtualMachineInstance with custom MAC address 
/root/go/src/kubevirt.io/kubevirt/tests/vmi_networking_test.go:220

[Fail] DataVolume Integration Starting a VirtualMachineInstance with a DataVolume as a volume source using Alpine import [It] should be successfully started and stopped multiple times 
/root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:62

[Fail] DataVolume Integration [rfe_id:896][crit:high][vendor:cnv-qe@redhat.com][level:system] with oc/kubectl [It] [test_id:836] Creating a VM with DataVolumeTemplates should succeed. 
/root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:228

[Fail] DataVolume Integration [rfe_id:896][crit:high][vendor:cnv-qe@redhat.com][level:system] with oc/kubectl [It] [test_id:837]deleting VM with cascade=true should automatically delete DataVolumes and VMI owned by VM. 
/root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:228

[Fail] DataVolume Integration [rfe_id:896][crit:high][vendor:cnv-qe@redhat.com][level:system] with oc/kubectl [It] [test_id:838]deleting VM with cascade=false should orphan DataVolumes and VMI owned by VM. 
/root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:228

[Fail] DataVolume Integration Starting a VirtualMachine with a DataVolume using Alpine import [It] should be successfully started and stopped multiple times 
/root/go/src/kubevirt.io/kubevirt/tests/utils.go:3326

Ran 315 of 340 Specs in 15470.756 seconds
FAIL! -- 284 Passed | 31 Failed | 1 Pending | 24 Skipped
```

**Comment by cynepco3hahue** [comment]
@SchSeba Do you have an idea why all multus tests fail?

**Comment by davidvossel** [comment]
i tested out the provider and was at least able to get it up and start a VMI on it. This is a good starting point.

File: cluster/ephemeral-provider-common.sh (hunk 1 above)
Comment (rmohr) [reply cluster/ephemeral-provider-common.sh]: It is needed for subsequent cluster-sync runs. Now with this change merged it only works on the first sync.

**Pull request merged**